src/port/port.tcl

```diff
@@ -2798,6 +2798,20 @@ proc action_selfupdate { action portlist opts } {
         fatal "port selfupdate failed: $result"
     }
 
+    if {![info exists options(ports_${action}_no-sync)] || !$options(ports_${action}_no-sync)} {
+        if {[dict get $selfupdate_status needed_portindex]} {
```

Comment:
Does `needed_portindex` actually required if the port base has been updated with no `--no-sync` given?

Seems that we can check `base_updated` at here to print messages for rerun `port selfupdate`.

Reply:
Moving this message to port.tcl looks good to me, as it is up to the client to restart the selfupdate (or ask the user to do so).

The check for --no-sync is the same as inside the macports1.0 API and that seems redundant. The `needed_portindex` would not be set otherwise.

But your change lost the condition on `base_updated` here.

As a side note, the macports1.0 API change is not backwards compatible, so we should only ship this with the next 2.x.0 release.

Reply:
> The check for --no-sync is the same as inside the macports1.0 API and that seems redundant. The `needed_portindex` would not be set otherwise.

Checking for `--no-sync` might be required for messages about restarting the selfupdate at here.
It can be returned by `macports::selfupdate`, but I'm not sure if it's necessary since there are some similar checks in `port.tcl`.


> But your change lost the condition on `base_updated` here.

The check for `base_updated` might not be required at here, as`portindex` may not running in some cases:

https://github.com/macports/macports-base/blob/87a8c78fe9f4eb5894c4a46134d5db49f14d22ed/src/macports1.0/macports.tcl#L3034

Reply:
This is when the ports tree was downloaded, but the Portfiles use new features of base, so they cannot be parsed with the old version. The message being printed only makes sense when base was updated.